pyActigraphy/io/bba/bba.py

```diff
@@ -4,8 +4,39 @@
 import re
 
 from ..base import BaseRaw
-from accelerometer.utils import date_parser
-from accelerometer.summarisation import imputeMissing
+from pandas.tseries.frequencies import to_offset
+#from accelerometer.utils import date_parser
+#from accelerometer.summarisation import imputeMissing
+
+# Native implementation of date_parser and imputeMissing
+def date_parser(t):
+    """
+    Adapted from the accelerometer package (utils.py)
+    """
+    tz = re.search(r'(?<=\[).+?(?=\])', t)
+    if tz is not None:
+        tz = tz.group()
+    t = re.sub(r'\[(.*?)\]', '', t)
+    return pd.to_datetime(t, utc=True).tz_convert(tz)
+
+def imputeMissing(data, extrapolate=True):
```

Comment:
At the moment, this function does not impute missing data but rather pad the data boundaries so that it covers a full day (24h). It misses the fillna function and its application (see https://github.com/OxWearables/biobankAccelerometerAnalysis/blob/ed0caad712a9686deb2bf3a37273c1dedd14beb6/src/accelerometer/summarisation.py#L250).

Reply:
I've now included the function `fillna`.

Reply:
@naja23 thanks for adding `fillna`. However it still misses the rest of the function from (https://github.com/OxWearables/biobankAccelerometerAnalysis/blob/ed0caad712a9686deb2bf3a37273c1dedd14beb6/src/accelerometer/summarisation.py#L264) up to `return data`.

Reply:
I think the function is now completely added! Also, imported numpy as it is required for fillna to work. Thanks for your assistance!

Reply:
Thanks for the changes. As mentioned earlier, I started a [discussion](https://github.com/OxWearables/biobankAccelerometerAnalysis/discussions/266) with the `accelerometer` package developers. Let see how it goes before moving on the PR.